MonoGame.Framework/BoundingSphere.cs

```diff
@@ -4,55 +4,73 @@
 
 using System;
 using System.Collections.Generic;
-using System.Globalization;
-using System.Runtime.Serialization;
 using System.Diagnostics;
+using System.Runtime.Serialization;
 
 namespace Microsoft.Xna.Framework
 {
-
+    /// <summary>
+    /// Describes a virtual sphere in 3D-space for bounding operations.
+    /// </summary>
     [DataContract]
     [DebuggerDisplay("{DebugDisplayString,nq}")]
     public struct BoundingSphere : IEquatable<BoundingSphere>
     {
         #region Public Fields
 
+        /// <summary>
+        /// The sphere center.
+        /// </summary>
         [DataMember]
         public Vector3 Center;
 
+        /// <summary>
+        /// The sphere radius.
+        /// </summary>
         [DataMember]
         public float Radius;
 
-        #endregion Public Fields
+        #endregion
+
+        #region Internal Properties
+
+        internal string DebugDisplayString
+        {
+            get
+            {
+                return string.Concat(
+                    "Center( ", this.Center.DebugDisplayString, " )  \r\n",
+                    "Radius( ", this.Radius.ToString(), " )"
+                    );
+            }
+        }
 
+        #endregion
 
         #region Constructors
 
+        /// <summary>
+        /// Constructs a bounding sphere with the specified center and radius.  
+        /// </summary>
+        /// <param name="center">The sphere center.</param>
+        /// <param name="radius">The sphere radius.</param>
         public BoundingSphere(Vector3 center, float radius)
         {
             this.Center = center;
             this.Radius = radius;
         }
 
-        #endregion Constructors
-
+        #endregion
 
         #region Public Methods
 
-        public BoundingSphere Transform(Matrix matrix)
-        {
-            BoundingSphere sphere = new BoundingSphere();
-            sphere.Center = Vector3.Transform(this.Center, matrix);
-            sphere.Radius = this.Radius * ((float)Math.Sqrt((double)Math.Max(((matrix.M11 * matrix.M11) + (matrix.M12 * matrix.M12)) + (matrix.M13 * matrix.M13), Math.Max(((matrix.M21 * matrix.M21) + (matrix.M22 * matrix.M22)) + (matrix.M23 * matrix.M23), ((matrix.M31 * matrix.M31) + (matrix.M32 * matrix.M32)) + (matrix.M33 * matrix.M33)))));
-            return sphere;
-        }
-
-        public void Transform(ref Matrix matrix, out BoundingSphere result)
-        {
-            result.Center = Vector3.Transform(this.Center, matrix);
-            result.Radius = this.Radius * ((float)Math.Sqrt((double)Math.Max(((matrix.M11 * matrix.M11) + (matrix.M12 * matrix.M12)) + (matrix.M13 * matrix.M13), Math.Max(((matrix.M21 * matrix.M21) + (matrix.M22 * matrix.M22)) + (matrix.M23 * matrix.M23), ((matrix.M31 * matrix.M31) + (matrix.M32 * matrix.M32)) + (matrix.M33 * matrix.M33)))));
-        }
+        #region Contains
 
+        /// <summary>
+        /// Gets how the specified <see cref="BoundingBox"/> lie within the bounds of this sphere.

```

Comment:
`Test if a bounding box is fully inside, outside, or just intersecting the sphere.`